docs/ref-manual/cvl/methods.md

```diff
@@ -46,44 +55,251 @@ method_summary   ::= "ALWAYS" "(" value ")"
+obliterates all knowlege that the prover has about the state of the contract
+before the call.
+
+The `HAVOC_ALL` approximation is {term}`sound`, but it can be overly
+restrictive in practice.  In reality, a contract's state cannot be changed in
+arbitrary ways, but only according to the contract's methods.  However, the
+Prover does not currently have support for more fine-grained reasoning about
+the side effects of unknown methods.
+
+A useful middle ground is the `HAVOC_ECF` summary type.  A `HAVOC_ECF`
+summarization for a method encodes the assumption that the called method is not
+reentrant.  This summarization approximates a method call by assuming it can
+have arbitrary effects on contracts other than the contract being verified, but
+that it can neither change the current contract's state nor decrease its ETH
+balance (aside from value transferred by the method call itself).
+
+The prover makes no assumptions about the return value of a havoc summary.  For
+methods that return multiple values, the approximations are allowed to return
+the incorrect number of results.  In most cases, this will cause the calling
+method to revert.  If you want to ignore this particular revert condition, you
+can pass the {ref}`-optimisticReturnsize` option.
+
+(dispatcher)=
+### `DISPATCHER` summaries
+
+The `DISPATCHER` summary type provides a useful approximation for methods of
+interfaces that are implemented by multiple contracts.  For example, the
+methods defined by the ERC20 specification are often summarized using the
+`DISPATCHER` summary type.
+
+If a function with a `DISPATCHER` summary is called, the Prover will assume
+that the receiver of the call is one of the known contract implementations
+containing the given signature; the call will then behave the same way that a
+normal method call on the receiver would.  The prover will consider examples
+with every possible implementing contract, but multiple `DISPATCHER` method
+calls on the same receiver address in the same example will use the same
+receiver contract.
+
+The set of contract implementations that the prover chooses from contains
+the set of contracts passed as [arguments to the CLI](../cli/options).
+In addition, the prover may consider an unknown target contract whose methods
+are all interpreted using the {ref}`AUTO summary <auto-summary>`.  The presence
+of the unknown contract is determined by the optional boolean argument to the
+`DISPATCHER` summary:
+
+ * With `DISPATCHER(false)` or just `DISPATCHER`, the unknown contract is
+   considered as a possiblity
+
+ * With `DISPATCHER(true)`, only the known contract instances are considered
+
+The most commonly used option is `DISPATCHER(true)`, because in most cases the
+behavior of `DISPATCHER(false)` is equivalent to that of `AUTO`.
+`DISPATCHER(false)` only behaves differently from `AUTO` when one of the known
+contracts exhibits behavior that `AUTO` assumes is impossible (such as
+reentrancy).
```

Comment:
I don't think this is true? If the callee address is *always* one of the known contracts (as proven by z3) then the two dispatcher flavors are equivalent? Maybe you're making a point about possible static behaviors...

Reply:
I weakened the claim in the current draft to something that I now think is true.